shell/platform/darwin/macos/framework/Source/FlutterAccessibilityMac.mm

```diff
@@ -0,0 +1,575 @@
+        std::u16string deleted_text;
+        std::u16string inserted_text;
+        id edit_text_marker = nil;
+        computeTextEdit(deleted_text, inserted_text, &edit_text_marker);
+        NSDictionary* user_info = GetUserInfoForValueChangedNotification(
+            native_node, deleted_text, inserted_text, edit_text_marker);
+        if (user_info) {
+          FireNativeMacNotificationWithUserInfo(native_node, mac_notification, user_info);
+          FireNativeMacNotificationWithUserInfo(
+              GetBridge()->GetFlutterAccessibilityFromID(kRootNode)->GetNativeViewAccessible(),
+              mac_notification, user_info);
+        }
+        return;
+      }
+      break;
+    case ax::AXEventGenerator::Event::LIVE_REGION_CREATED:
+      mac_notification = NSAccessibilityLiveRegionCreatedNotification;
+      break;
+    case ax::AXEventGenerator::Event::ALERT: {
+      FireNativeMacNotification(native_node, NSAccessibilityLiveRegionCreatedNotification);
+      // Voiceover requires a live region changed notification to actually
+      // announce the live region.
+      AXEventGenerator::EventParams event_params(ax::AXEventGenerator::Event::LIVE_REGION_CHANGED,
+                                                 ax::EventFrom::kNone, {});
+      AXEventGenerator::TargetedEvent live_region(GetAXNode(), event_params);
+      OnAccessibilityEvent(live_region);
+      return;
+    }
+    case ax::AXEventGenerator::Event::LIVE_REGION_CHANGED: {
+      if (SystemVersionLessOrEqualTo({.majorVersion = 10, .minorVersion = 13})) {
+        // Use the announcement API to get around OS <= 10.13 VoiceOver bug
+        // where it stops announcing live regions after the first time focus
+        // leaves any content area.
+        // Unfortunately this produces an annoying boing sound with each live
+        // announcement, but the alternative is almost no live region support.
+        NSString* announcement = [[NSString alloc] initWithUTF8String:GetLiveRegionText().c_str()];
+        NSDictionary* notification_info = @{
+          NSAccessibilityAnnouncementKey : announcement,
+          NSAccessibilityPriorityKey : @(NSAccessibilityPriorityLow)
+        };
+        // Trigger VoiceOver speech and show on Braille display, if available.
+        // The Braille will only appear for a few seconds, and then will be replaced
+        // with the previous announcement.
+        FireNativeMacNotificationWithUserInfo([NSApp mainWindow],
+                                              NSAccessibilityAnnouncementRequestedNotification,
+                                              notification_info);
+        return;
+      }
+
+      // Use native VoiceOver support for live regions.
+
+      std::async(std::launch::async, [&native_node, this]() {
+        // Use sleep_for to wait specified time (or sleep_until).
+        std::this_thread::sleep_for(std::chrono::milliseconds{kLiveRegionChangeIntervalMS});
```

Comment:
Please avoid thread sleeps.

Also, the comment above this line is not especially helpful - it's just saying we're calling a function that does a sleep.

What happens if we fire the notification right away?

What happens if we're on a slower/loaded machine if 25ms is not enough to avoid that?